modification_request.go

```diff
@@ -0,0 +1,66 @@
+package braintree
+
+import (
+	"encoding/xml"
+)
+
+type ModificationRequest struct {
+	Amount                *Decimal `xml:"amount,omitempty"`
+	NumberOfBillingCycles int      `xml:"number-of-billing-cycles,omitempty"`
+	Quantity              int      `xml:"quantity,omitempty"`
+	NeverExpires          bool     `xml:"never-expires,omitempty"`
+}
+
+type AddModificationRequest struct {
+	ModificationRequest
+	InheritedFromID string `xml:"inherited-from-id,omitempty"`
+}
+
+type UpdateModificationRequest struct {
+	ModificationRequest
+	ExistingID string `xml:"existing-id,omitempty"`
+}
+
+type ModificationsRequest struct {
+	Add               []AddModificationRequest
+	Update            []UpdateModificationRequest
+	RemoveExistingIDs []string
+}
+
+func (m ModificationsRequest) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
```

Comment:
@leighmcculloch , this definitely works. I wonder whether it will be simpler to use another struct like I did in my pr. In that struct you can set the Type="array". That struct has to be a public struct but you can put that in an internal pkg to make sure outside pkgs can't use it. I have not tried in code.
https://github.com/lionelbarrow/braintree-go/pull/94/files#diff-1dd96d11be5f0821e43a50897a292b4dR42